### ISSUE TITLE
I've got following error from catkin_make

### ISSUE THREAD

**Issue by limhyon** [turtlebot/turtlebot_arm#10]
I've got following error from catkin_make.
How do I solve this?

[100%] Building CXX object turtlebot_arm/turtlebot_block_manipulation/CMakeFiles/block_detection_action_server.dir/src/block_detection_action_server.cpp.o
In file included from /opt/ros/groovy/include/class_loader/class_loader_core.h:39:0,
                 from /opt/ros/groovy/include/class_loader/class_loader_register_macro.h:33,
                 from /opt/ros/groovy/include/class_loader/class_loader.h:39,
                 from /opt/ros/groovy/include/pluginlib/class_list_macros.h:40,
                 from /home/hyon/catkin_ws/src/turtlebot_arm/turtlebot_arm_ikfast_plugin/src/turtlebot_arm_arm_ikfast_moveit_plugin.cpp:830:
/opt/ros/groovy/include/class_loader/meta_object.h: In member function ‘B\* class_loader::class_loader_private::MetaObject<C, B>::create() const [with C = ikfast_kinematics_plugin::IKFastKinematicsPlugin, B = kinematics::KinematicsBase]’:
/home/hyon/catkin_ws/src/turtlebot_arm/turtlebot_arm_ikfast_plugin/src/turtlebot_arm_arm_ikfast_moveit_plugin.cpp:831:101:   instantiated from here
/opt/ros/groovy/include/class_loader/meta_object.h:193:17: error: cannot allocate an object of abstract type ‘ikfast_kinematics_plugin::IKFastKinematicsPlugin’
/home/hyon/catkin_ws/src/turtlebot_arm/turtlebot_arm_ikfast_plugin/src/turtlebot_arm_arm_ikfast_moveit_plugin.cpp:64:7: note:   because the following virtual functions are pure within ‘ikfast_kinematics_plugin::IKFastKinematicsPlugin’:
/opt/ros/groovy/include/moveit/kinematics_base/kinematics_base.h:90:16: note:   virtual bool kinematics::KinematicsBase::getPositionIK(const Pose&, const std::vector<double>&, std::vector<double>&, moveit_msgs::MoveItErrorCodes&, const kinematics::KinematicsQueryOptions&) const
/opt/ros/groovy/include/moveit/kinematics_base/kinematics_base.h:108:16: note:  virtual bool kinematics::KinematicsBase::searchPositionIK(const Pose&, const std::vector<double>&, double, std::vector<double>&, moveit_msgs::MoveItErrorCodes&, const kinematics::KinematicsQueryOptions&) const
/opt/ros/groovy/include/moveit/kinematics_base/kinematics_base.h:128:16: note:  virtual bool kinematics::KinematicsBase::searchPositionIK(const Pose&, const std::vector<double>&, double, const std::vector<double>&, std::vector<double>&, moveit_msgs::MoveItErrorCodes&, const kinematics::KinematicsQueryOptions&) const
/opt/ros/groovy/include/moveit/kinematics_base/kinematics_base.h:150:16: note:  virtual bool kinematics::KinematicsBase::searchPositionIK(const Pose&, const std::vector<double>&, double, std::vector<double>&, const IKCallbackFn&, moveit_msgs::MoveItErrorCodes&, const kinematics::KinematicsQueryOptions&) const
/opt/ros/groovy/include/moveit/kinematics_base/kinematics_base.h:173:16: note:  virtual bool kinematics::KinematicsBase::searchPositionIK(const Pose&, const std::vector<double>&, double, const std::vector<double>&, std::vector<double>&, const IKCallbackFn&, moveit_msgs::MoveItErrorCodes&, const kinematics::KinematicsQueryOptions&) const
make[2]: **\* [turtlebot_arm/turtlebot_arm_ikfast_plugin/CMakeFiles/turtlebot_arm_moveit_ikfast_kinematics_plugin.dir/src/turtlebot_arm_arm_ikfast_moveit_plugin.cpp.o] Error 1
make[1]: **\* [turtlebot_arm/turtlebot_arm_ikfast_plugin/CMakeFiles/turtlebot_arm_moveit_ikfast_kinematics_plugin.dir/all] Error 2
make[1]: **\* Waiting for unfinished jobs....
Linking CXX executable /home/hyon/catkin_ws/devel/lib/turtlebot_block_manipulation/block_detection_action_server
Linking CXX executable /home/hyon/catkin_ws/devel/lib/turtlebot_kinect_arm_calibration/calibrate_kinect_checkerboard
[100%] [100%] Built target block_detection_action_server
Built target calibrate_kinect_checkerboard
make: **\* [all] Error 2
Invoking "make" failed
hyon@hyon-gt683:~/catkin_ws$ sudo apt-get install ros-groovy-^C
hyon@hyon-gt683:~/catkin_ws$

**Comment by mikeferguson** [comment]
The interface of moveit plugins has been changed (there is a thread on the moveit mailing list describing it). The plugin needs to be regenerated and/or hacked.

Also, just a heads up -- this repo is a work in progress -- I only got as far as the configuration stuff and running some simple commands from MoveIt before it ended up falling off my todo list for a while. I have not tested calibration or blocks, and would not expect them to work -- if I remember correctly, not even all targets are building.
